packages/web/src/services/solana/solana.ts

```diff
@@ -143,6 +144,25 @@ export const getUSDCAssociatedTokenAccount = async (
   )
 }
 
+/**
+ * Returns the owner of the USDC token acccount, if the provided account
+ * is a USDC token account. Otherwise, just returns the account
+ */
+export const getUSDCAssociatedTokenAccountOwner = async (
+  accountAddress: SolanaWalletAddress
+) => {
+  const connection = await getSolanaConnection()
+  const isAta = await isTokenAccount({ accountAddress, mint: 'usdc' })
```

Comment:
total nit, but this will end up hitting the RPC twice - once to verify that it's a token account, and another to fetch it again and get the owner. We could probably just try/catch `getAccount` and return the owner if it doesn't throw

Reply:
going to do this, but also add a check for the mint then

Reply:
Honestly would prefer a method that just gets the token account owner over one that specifically does so for USDC only - why do we need the mint check here?

Reply:
doesn't matter, you're right